.circleci/config.yml

```diff
@@ -1,49 +1,48 @@
 ---
 version: 2.1
 
-shared-linux: &shared-linux
-  working_directory: /go/src/github.com/prometheus/procfs
-  steps:
-  - checkout
-  - run: make style check_license vet test staticcheck
-
-shared-windows: &shared-windows
-  working_directory: /go/src/github.com/prometheus/procfs
-  steps:
-  - checkout
-  - run: make style check_license vet staticcheck
+commands:
+  linux:
+    steps:
+    - run: make style check_license vet test staticcheck
+  windows:
+    steps:
+    - run: make style check_license vet staticcheck
 
 jobs:
-  test-linux-1-10:
-    <<: *shared-linux
-    environment:
-      GOOS: linux
-    docker:
-    - image: circleci/golang:1.10
-  test-linux-1-11:
-    <<: *shared-linux
-    environment:
-      GOOS: linux
+  test:
+    parameters:
+      go_version:
+        type: string
+      os:
+        type: string
     docker:
-    - image: circleci/golang:1.11
-  test-windows-1-10:
-    <<: *shared-windows
+    - image: circleci/golang:<< parameters.go_version >>
     environment:
-      GOOS: windows
-    docker:
-    - image: circleci/golang:1.10
-  test-windows-1-11:
-    <<: *shared-windows
-    environment:
-      GOOS: windows
-    docker:
-    - image: circleci/golang:1.11
+      GOOS: << paramaters.os >>
+    working_directory: /go/src/github.com/prometheus/procfs
+    steps:
+    - checkout
+    - run: go env
+    - << parameters.os >>
 
 workflows:
   version: 2
-  node_exporter:
+  procfs:
     jobs:
-    - test-linux-1-10
-    - test-linux-1-11
-    - test-windows-1-10
-    - test-windows-1-11
+    - test:
+        name: linux-1-10
+        os: linux
+        go_version: "1.10"
+    - test:
+        name: linux-1-11
+        os: linux
+        go_version: "1.11"
+    - test:
+        name: windows-1-10
+        os: windows
```

Comment:
Why do we run procfs tests on Windows? That OS isn't supported. When was this introduced?

Reply:
Due to package requirements, client_golang requires procfs. There's basically one piece of code in client_golang that causes all of procfs to get pulled in as a requirement. I've complained about this several times, but no progress has been made to clean it up.

https://github.com/prometheus/procfs/pull/112

Reply:
Well, that part of client_golang was the reason I created the procfs repo :-) The reason that we pull in even sysfs and all the ne packages created in the procfs repo is something I would like to change. There is no need that we couple all these packages together.

Thanks for the link to the pr.